### PR TITLE
Task/sdk 3710/support v6.1.0

### PR DESCRIPTION
**Pull request by Anush-Shand**
Android Support For v6.1.1

Also Upgrades Gradle to 8.2.2

### CODE REVIEW

File: Example/package.json [hunk 1]
```diff
@@ -10,18 +10,20 @@
     "lint": "eslint ."
   },
   "dependencies": {
+    "@react-native/metro-config": "^0.72.12",
```
Comment (CTLalit) [inline Example/package.json:13]: why is this added?

File: Example/android/app/build.gradle [hunk 2]
```diff
@@ -85,7 +85,6 @@ project.ext.react = [
     nodeExecutableAndArgs:  ["/usr/local/bin/node"]
 ]
 
-apply from: "../../node_modules/react-native/react.gradle"
```
Comment (CTLalit) [inline Example/android/app/build.gradle:88]: what did this contain? are they moved in this file?

Comment (Anush-Shand) [reply Example/android/app/build.gradle:88]: The path has been changed as specified in this doc https://react-native-community.github.io/upgrade-helper/?from=0.71.8&to=0.73.5

File: Example/package.json (hunk 1 above)
Comment (Anush-Shand) [reply Example/package.json:13]: Required to run example app after upgrading RN version

File: android/build.gradle [hunk 3]
```diff
@@ -8,22 +8,23 @@ buildscript {
...
-        classpath 'com.android.tools.build:gradle:4.0.1'
+        classpath 'com.android.tools.build:gradle:8.2.2'
     }
 }
 
 android {
-    compileSdkVersion 33
-    buildToolsVersion '33.0.0'
+    compileSdkVersion 34
+    buildToolsVersion '34.0.0'
```
Comment (piyush-kukadiya) [inline android/build.gradle:18]: @Anush-Shand Since we are bumping Android to 34 please cross check which react-native version we are working with, also note it down if anyone asks. Also cross check release notes of react-native version that we are using.

**Review by piyush-kukadiya — COMMENTED** [review]
Added comments

**Comment by darshanclevertap** [comment]
@Anush-Shand please update the description to mention update to v6.1.1

File: android/build.gradle (hunk 3 above)
Comment (Anush-Shand) [reply android/build.gradle:18]: @piyush-kukadiya     
"react-native": "^0.49.5"

This is react native version used and I don't think it has been updated for long. I think it works since we use very basic features of React-Native in the lib

Do we wish to update
The latest version is 0.73.5

Comment (shivamsharma2710) [reply android/build.gradle:18]: @Anush-Shand We may update to the latest either with this release or the coming release.

**Review by shivamsharma2710 — APPROVED** [review]
LGTM

File: Example/android/app/build.gradle (hunk 2 above)
Comment (CTLalit) [reply Example/android/app/build.gradle:88]: i did not see in doc. Test once and resolve if this is okay.

File: Example/metro.config.js [hunk 4]
```diff
@@ -1,17 +1,31 @@
...
-  },
+const { getDefaultConfig, mergeConfig } = require('@react-native/metro-config');
+
+const defaultConfig = getDefaultConfig(__dirname);
+
+const {
+ resolver: { sourceExts, assetExts },
+} = getDefaultConfig(__dirname);
+
+const config = {
```
Comment (CTLalit) [inline Example/metro.config.js:15]: make sure this config is correct

File: Example/android/app/build.gradle (hunk 2 above)
Comment (Anush-Shand) [reply Example/android/app/build.gradle:88]: This line is no longer needed as this file is removed in this commit 
https://github.com/facebook/react-native/commit/d4a9bdc40e06bdda9565cc43ea5af5a13ff6f1cf

File: android/build.gradle (hunk 3 above)
Comment (Anush-Shand) [reply android/build.gradle:18]: @shivamsharma2710 Decided to update to the latest along with the task to support new architecture in the future sprint

File: Example/metro.config.js (hunk 4 above)
Comment (Anush-Shand) [reply Example/metro.config.js:15]:
```
missing-asset-registry-path could not be found within the project or in these directories:
  node_modules/react-native/node_modules
  node_modules
  ../node_modules
  ../../node_modules
> 1 | �PNG
  2 | 
  3 | 
  4 | IHD��IDATxڝՅ
```

It was added to resolve this. Works fine

File: android/build.gradle (hunk 3 above)
Comment (shivamsharma2710) [reply android/build.gradle:18]: @Anush-Shand Sure.
